src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,22 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        UserData userData = new UserData(userId, "Details for user" + userId);
```

Comment:
The string concatenation in 'new UserData(userId, "Details for user" + userId)' lacks a space before the userId. It should be corrected to ensure proper formatting of the details string.